components/token-container.tsx

```diff
@@ -18,11 +21,44 @@ type TokenContainerProps = FlexProps & {
   setAmount: Dispatch<SetStateAction<string>>;
   setFocused: Dispatch<SetStateAction<string>>;
   maxBalance?: string;
+  calculatePriceAndToken?: () => void;
 };
 
 export const TokenContainer = (props: TokenContainerProps) => {
   const ethCard = props.symbol === "ETH";
 
+  const maxValueWithGas = async () => {
+    if (!props.maxBalance || parseFloat(props.maxBalance) === 0) return "0";
+
+    const defaultGasEstimate = parseEther("0.0045");
+    const valueAsBigNumber = parseEther(props.maxBalance);
+    const maxGasUsage = 100000; // Max gas usage seen on Etherscan was ~85k
+
+    try {
+      const feeData = await estimateFeesPerGas(mainnetClient);
+
+      if (feeData) {
+        if (feeData.maxFeePerGas) {
+          const maxFee = formatEther(feeData.maxFeePerGas);
+          const gasPriceEther = parseFloat(maxFee);
+          if (!isNaN(gasPriceEther)) {
+            let gasCost = parseEther((gasPriceEther * maxGasUsage).toString());
+            if (gasCost < defaultGasEstimate) {
+              gasCost = defaultGasEstimate;
+            }
+            const valMinusGas = valueAsBigNumber - gasCost;
+            if (valMinusGas <= 0n) return "0";
+            return formatEther(valMinusGas);
+          }
+        }
+      }
+    } catch (error) {}
```

Comment:
you shoud log error so that we see it in vercel

Reply:
Done